packages/transaction-controller/src/helpers/EtherscanRemoteTransactionSource.ts

```diff
@@ -33,6 +34,10 @@ export class EtherscanRemoteTransactionSource
 
   #isTokenRequestPending: boolean;
 
+  #mutex = new Mutex();
+
+  API_FETCH_INTERVAL = 5000;
```

Comment:
perhaps another name like `ETHERSCAN_RATE_LIMIT_INTERVAL` would be more self describing of what the number represents? Or at least it would make it more obvious without having to scan the mutex logic below first

Reply:
nit

Reply:
done here: https://github.com/MetaMask/core/pull/3822/commits/00f5192cff537500fa217bd755e90f733746d878